File: src/models/responses/PostDetails.ts
import { UserHeader } from "./UserHeader";

export interface Post {
    id: string,
    content: string,
    description: string,
    areDisabledComments: boolean,
    creationDatetime: string,
    areHiddenLikes: boolean,
    author: UserHeader,
    likesCount: number,
    commentsCount: number,
    didLoggedUserLikePost: boolean
}

File: code/instagram/backend/src/main/java/pl/instagram/instagram/model/api/response/PostDetails.java
package pl.instagram.instagram.model.api.response;

import java.time.OffsetDateTime;

public record PostDetails(
    String id,
    OffsetDateTime creationDatetime,
    String description,
    String content,
    boolean areHiddenLikes,
    boolean areDisabledComments,
    UserHeader author,
    Integer likesCount,
    Integer commentsCount,
    boolean didLoggedUserLikePost
){
    public PostDetails(String id, OffsetDateTime creationDatetime, String description, String content, boolean areHiddenLikes, boolean areDisabledComments, UserHeader author, Integer likesCount, Integer commentsCount) {
        this(id, creationDatetime, description, content, areHiddenLikes, areDisabledComments, author, likesCount, commentsCount, false);
    }
}
